crypto/threads_pthread.c

```diff
@@ -196,6 +196,19 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
 
 # ifndef FIPS_MODULE
 #  ifdef OPENSSL_SYS_UNIX
+
+void OPENSSL_fork_prepare(void)
+{
+}
+
+void OPENSSL_fork_parent(void)
+{
+}
+
+void OPENSSL_fork_child(void)
+{
+}
+
 static pthread_once_t fork_once_control = PTHREAD_ONCE_INIT;
 
 static void fork_once_func(void)
```

Comment:
The `pthread_atfork` code will also have to have `#ifndef OPENSSL_NO_DEPRECATED_3_0` around it so that `fork_once_func` does nothing in the event of a `no-deprecated` build.